pkg/odo/cli/dev/dev.go

```diff
@@ -268,3 +310,105 @@ It forwards endpoints with any exposure values ('public', 'internal' or 'none')
+			} else {
+				errors = append(errors, fmt.Sprintf("container %q not found in the devfile", fPort.ContainerName))
+			}
+		} else {
+			// Validate that a custom port mapping for port forwarding config without a container targets a unique containerPort
+			if containers, ok := portContainerMapping[fPort.ContainerPort]; ok && len(containers) > 1 {
+				// this workaround makes sure the unit tests do not fail because of a change in the order
+				sort.Strings(containers)
+				msg := fmt.Sprintf("multiple container components (%s) found with same container port %d in the devfile, port forwarding must be defined with format <localPort>:<containerName>:<containerPort>", strings.Join(containers, ", "), fPort.ContainerPort)
+				if !strings.Contains(strings.Join(errors, ", "), msg) {
+					errors = append(errors, msg)
+				}
+
+				continue portLoop
+			}
+			for _, containerEndpoints := range containerEndpointMapping {
+				for _, endpoint := range containerEndpoints {
+					if endpoint.TargetPort == fPort.ContainerPort {
+						klog.V(1).Infof("container port %d matches %s endpoints of container %s", fPort.ContainerPort, endpoint.Name, fPort.ContainerName)
+						continue portLoop
+					}
+				}
+			}
+			errors = append(errors, fmt.Sprintf("container port %d not found in the devfile container endpoints", fPort.ContainerPort))
+		}
+	}
+	if len(errors) != 0 {
+		return errors, fmt.Errorf("values for --port-forward flag are invalid")
+	}
+	return nil, nil
+}
+
+// parsePortForwardFlag parses custom port forwarding configuration; acceptable patterns: <localPort>:<containerPort>, <localPort>:<containerName>:<containerPort>
+func parsePortForwardFlag(portForwardFlag []string) (forwardedPorts []api.ForwardedPort, err error) {
+	// acceptable examples: 8000:runtime_123:8080, 8000:9000, 8000:runtime:8080, 20001:20000
+	// unacceptable examples: :8000, 80000:
+	pattern := `^(\d{1,5})(:\w*)?:(\d{1,5})$`
+	portReg := regexp.MustCompile(pattern)
+	const largestPortValue = 65535
+	for _, portData := range portForwardFlag {
+		if !portReg.MatchString(portData) {
+			return nil, errors.New("ports are not defined properly, acceptable formats are: <localPort>:<containerPort>, <localPort>:<containerName>:<containerPort>")
+		}
+		var portF api.ForwardedPort
+
+		portDataArr := strings.Split(portData, ":")
+		switch len(portDataArr) {
+		case 2:
+			portF.LocalPort, _ = strconv.Atoi(portDataArr[0])
+			portF.ContainerPort, _ = strconv.Atoi(portDataArr[1])
+		case 3:
+			portF.LocalPort, _ = strconv.Atoi(portDataArr[0])
+			portF.ContainerName = portDataArr[1]
+			portF.ContainerPort, _ = strconv.Atoi(portDataArr[2])
+		}
+		if !(portF.LocalPort > 0 && portF.LocalPort <= largestPortValue) || !(portF.ContainerPort > 0 && portF.ContainerPort <= largestPortValue) {
+			return nil, fmt.Errorf("%s is invalid; port number must not be greater than 0 and less than %d", portData, largestPortValue)
```

Comment:
Port numbers must actually be > 0 and <= 65535:

```suggestion
			return nil, fmt.Errorf("%s is invalid; port numbers must be greater than 0 and less than %d", portData, largestPortValue)
```